pensando_psm/utils.py

```diff
@@ -0,0 +1,232 @@
+    """Load global state from disk and unpickle"""
+    try:
+        config_id = config.get('config_id', 'generic')
+        psm_session_unique = f'{PSM_SESSION_FILE}_{config_id}'
+        psm_cookie_exp_unique = f'{PSM_COOKIE_EXP_FILE}_{config_id}'
+
+        with open(os.path.join(TMP_FILE_ROOT, psm_session_unique), 'rb') as file:
+            psm.session = pickle.load(file)
+
+        with open(os.path.join(TMP_FILE_ROOT, psm_cookie_exp_unique), 'rb') as file:
+            psm.cookie_expiration = pickle.load(file)
+
+        logger.info('Loaded session state successfully')
+
+    except Exception as ex:
+        logger.warning(f'Error loading session state: {ex}')
+
+
+def _set_state(config):
+    """Pickle global state and save to disk"""
+    try:
+        config_id = config.get('config_id', 'generic')
+        psm_session_unique = f'{PSM_SESSION_FILE}_{config_id}'
+        psm_cookie_exp_unique = f'{PSM_COOKIE_EXP_FILE}_{config_id}'
+
+        with open(os.path.join(TMP_FILE_ROOT, psm_session_unique), 'wb') as file:
+            # Pickle the Requests Session() object
+            pickle.dump(psm.session, file, pickle.HIGHEST_PROTOCOL)
+
+        with open(os.path.join(TMP_FILE_ROOT, psm_cookie_exp_unique), 'wb') as file:
+            # Pickle the cookie_expiration variable
+            pickle.dump(psm.cookie_expiration, file, pickle.HIGHEST_PROTOCOL)
+
+    except Exception as ex:
+        logger.exception(f'Error saving session state: {ex}')
+        raise ConnectorError(f'Error saving session state: {ex}')
+
+    logger.info('Saved session state successfully')
+
+
+def _debug_remove_session_state(config):
+    try:
+        config_id = config.get('config_id', 'generic')
+        psm_session_unique = f'{PSM_SESSION_FILE}_{config_id}'
+        psm_cookie_exp_unique = f'{PSM_COOKIE_EXP_FILE}_{config_id}'
+
+        os.remove(os.path.join(TMP_FILE_ROOT, psm_session_unique))
+        os.remove(os.path.join(TMP_FILE_ROOT, psm_cookie_exp_unique))
+        logger.info('Debug: Session state removed from disk')
+    except Exception as ex:
+        logger.warning(f'Debug: Error removing Session State from disk: {ex}')
+
+
+def _debug_reset_session_state(config):
+    try:
+        config_id = config.get('config_id', 'generic')
+        psm_session_unique = f'{PSM_SESSION_FILE}_{config_id}'
+        psm_cookie_exp_unique = f'{PSM_COOKIE_EXP_FILE}_{config_id}'
+
+        reset_session = requests.Session()
+        reset_cookie_expiration = None
+        with open(os.path.join(TMP_FILE_ROOT, psm_session_unique), 'wb') as file:
+            pickle.dump(reset_session, file, pickle.HIGHEST_PROTOCOL)
+
+        with open(os.path.join(TMP_FILE_ROOT, psm_cookie_exp_unique), 'wb') as file:
+            pickle.dump(reset_cookie_expiration, file, pickle.HIGHEST_PROTOCOL)
+        logger.info('Debug: Session state reset on disk')
+    except Exception as ex:
+        logger.warning(f'Debug: Error resetting Session State on disk: {ex}')
```

Comment:
```suggestion
        logger.debug(f'Error resetting Session State on disk: {ex}')
```

Reply:
This is a test operation that will be deleted.